### PR TITLE
Port recyclerview module to rxjava2

### PR DESCRIPTION
**Pull request by marcinsus**

### CODE REVIEW

File: rxbinding-recyclerview-v7/src/main/java/com/jakewharton/rxbinding/support/v7/widget/RecyclerAdapterDataChangeObservable.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+final class RecyclerAdapterDataChangeObservable<T extends Adapter<? extends ViewHolder>>
+    extends Observable<T> {
+  private final T adapter;
+
+  RecyclerAdapterDataChangeObservable(T adapter) {
+    this.adapter = adapter;
+  }
+
+  @Override protected void subscribeActual(Observer<? super T> observer) {
```
Comment (OrdonTeam) [inline rxbinding-recyclerview-v7/src/main/java/com/jakewharton/rxbinding/support/v7/widget/RecyclerAdapterDataChangeObservable.java]: verifyMainThread is missing

File: rxbinding-recyclerview-v7/src/main/java/com/jakewharton/rxbinding/support/v7/widget/RecyclerViewChildAttachStateChangeEventObservable.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import io.reactivex.android.MainThreadDisposable;
+
+final class RecyclerViewChildAttachStateChangeEventObservable extends Observable<RecyclerViewChildAttachStateChangeEvent> {
+  private final RecyclerView view;
+
+  RecyclerViewChildAttachStateChangeEventObservable(RecyclerView recyclerView) {
+    this.view = recyclerView;
+  }
+
+  @Override protected void subscribeActual(Observer<? super RecyclerViewChildAttachStateChangeEvent> observer) {
```
Comment (OrdonTeam) [inline rxbinding-recyclerview-v7/src/main/java/com/jakewharton/rxbinding/support/v7/widget/RecyclerViewChildAttachStateChangeEventObservable.java]: verifyMainThread is missing

File: rxbinding-recyclerview-v7/src/main/java/com/jakewharton/rxbinding/support/v7/widget/RecyclerViewScrollEventObservable.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import io.reactivex.android.MainThreadDisposable;
+
+final class RecyclerViewScrollEventObservable extends Observable<RecyclerViewScrollEvent> {
+  private final RecyclerView view;
+
+  RecyclerViewScrollEventObservable(RecyclerView view) {
+    this.view = view;
+  }
+
+  @Override protected void subscribeActual(Observer<? super RecyclerViewScrollEvent> observer) {
```
Comment (OrdonTeam) [inline rxbinding-recyclerview-v7/src/main/java/com/jakewharton/rxbinding/support/v7/widget/RecyclerViewScrollEventObservable.java]: verifyMainThread is missing

File: rxbinding-recyclerview-v7/src/main/java/com/jakewharton/rxbinding/support/v7/widget/RecyclerViewChildAttachStateChangeEventObservable.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    Listener listener = new Listener(view, observer);
+    observer.onSubscribe(listener);
+    view.addOnChildAttachStateChangeListener(listener);
+  }
+
+  final class Listener extends MainThreadDisposable implements OnChildAttachStateChangeListener {
+    private final RecyclerView recyclerView;
+    private Observer<? super RecyclerViewChildAttachStateChangeEvent> observer;
+
+    Listener(final RecyclerView recyclerView, final Observer<? super RecyclerViewChildAttachStateChangeEvent> observer) {
```
Comment (OrdonTeam) [inline rxbinding-recyclerview-v7/src/main/java/com/jakewharton/rxbinding/support/v7/widget/RecyclerViewChildAttachStateChangeEventObservable.java]: remove finals

File: rxbinding-recyclerview-v7/src/main/java/com/jakewharton/rxbinding/support/v7/widget/RecyclerViewScrollStateChangeObservable.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+import io.reactivex.android.MainThreadDisposable;
+
+final class RecyclerViewScrollStateChangeObservable extends Observable<Integer> {
+  private final RecyclerView view;
+
+  RecyclerViewScrollStateChangeObservable(RecyclerView view) {
+    this.view = view;
+  }
+
+  @Override protected void subscribeActual(Observer<? super Integer> observer) {
```
Comment (OrdonTeam) [inline rxbinding-recyclerview-v7/src/main/java/com/jakewharton/rxbinding/support/v7/widget/RecyclerViewScrollStateChangeObservable.java]: verifyMainThread is missing

**Comment by marcinsus** [comment]
Done

**Pull request merged**